File: app/src/main/java/com/xpf/p2p/base/MvpBasePresenter.java
package com.xpf.p2p.base;

import java.lang.ref.WeakReference;

/**
 * Created by xpf on 2018/5/4 :)
 * Function:
 */
public class MvpBasePresenter<T> {

    protected WeakReference<T> mViewRef;

    /**
     * attach view reference
     *
     * @param view
     */
    public void attachView(T view) {
        mViewRef = new WeakReference<>(view);
    }

    /**
     * detach view reference
     */
    public void detachView() {
        mViewRef.clear();
    }

    /**
     * is or not nullable
     */
    protected boolean isNonNull() {
        return mViewRef.get() != null;
    }

    /**
     * get view
     */
    protected T getView() {
        return mViewRef.get();
    }
}

File: app/src/main/java/com/xpf/p2p/base/MvpBasePresenter.kt
package com.xpf.p2p.base

import java.lang.ref.WeakReference

/**
 * MVP Presenter 基类（Kotlin 版本）
 * 保留用于尚未迁移到 MVVM 的模块
 */
open class MvpBasePresenter<T> {

    protected var mViewRef: WeakReference<T>? = null

    fun attachView(view: T) {
        mViewRef = WeakReference(view)
    }

    fun detachView() {
        mViewRef?.clear()
    }

    protected fun isNonNull(): Boolean = mViewRef?.get() != null

    protected fun getView(): T? = mViewRef?.get()
}
